File: web/src/main/webapp/resources/client/Pms/modules/bankDetails/model/BankDetails.js
Ext.define("Pms.modules.bankDetails.model.BankDetails", {
    extend: "Ext.data.Model",
    alias: 'widget.bankDetailsModel',

    fields: [
        'id',
        'name',
        Pms.stringFieldUseNull('holder'),
        {name: 'bankName', useNull: true},
        {name: 'account', type: 'int', useNull: true},
        {name: 'edrpou', type: 'int', useNull: true},
        {name: 'mfo', type: 'int', useNull: true},
        {name: 'defaultDetails', persist: false},
        {name: 'blocked', persist: false},
        {name: 'description', useNull: true},
        'paymentType',
        Pms.stringFieldUseNull('swift'),
        Pms.stringFieldUseNull('bankAddress'),
        Pms.stringFieldUseNull('corrBankName'),
        Pms.stringFieldUseNull('corrAccount'),
        Pms.stringFieldUseNull('corrSwift'),
        Pms.stringFieldUseNull('corrBankAddress'),
        {name: 'cardNumber', type: 'int', useNull: 'true'},
        Pms.stringFieldUseNull('additional'),
        Pms.stringFieldUseNull('iban')
    ]
});

File: domain/src/main/java/com/idgindigo/pms/domain/pms/BankDetails.java
package com.idgindigo.pms.domain.pms;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.idgindigo.pms.domain.BaseEntity;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;

/**
 * @author valentyn_vakatsiienko
 * @since 12/30/13 12:36 PM
 */
@Entity
@Getter
@Setter
public class BankDetails extends BaseEntity {
    public static final String BANK_DETAILS = "bankDetails";

    @NotNull
    private String name;

    private String bankName;
    @Pattern(regexp = "^[0-9]*$")
    private String account;
    private String swift;
    @Column(length = 1000)
    private String bankAddress;

    private String corrBankName;
    @Pattern(regexp = "^[0-9]*$")
    private String corrAccount;
    private String corrSwift;
    @Column(length = 1000)
    private String corrBankAddress;
    private String iban;

    private Long edrpou;
    private Long mfo;
    private Long cardNumber;
    private String holder;

    private String description;
    @Column(length = 1000)
    private String additional;
    @NotNull
    private PaymentType paymentType;
    @Column(updatable = false)
    @JsonIgnore
    @Getter(onMethod = @_({@JsonProperty}))
    @Setter(onMethod = @_({@JsonIgnore}))
    private Boolean defaultDetails = Boolean.FALSE;
    @Column(nullable = false)
    private Boolean blocked = Boolean.FALSE;

    public enum PaymentType {
        CARD, CASH, TRANSFER, INTERNATIONAL_ACCOUNT
    }

}
